monix-reactive/shared/src/main/scala/monix/reactive/observables/CombineObservable.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.observables
+
+import cats.Apply
+import monix.reactive.Observable
+
+/** A `CombineObservable` is an observable that wraps a regular
+  * [[Observable]] and provide [[cats.Apply]] instance
+  * which uses [[Observable.combineLatest]] to combine elements.
+  */
+final class CombineObservable[A](val value: Observable[A]) extends AnyVal
```

Comment:
This is OK-ish, but I'd like to avoid any boxing, plus I think we should be standardizing on a newtype encoding in the Cats ecosystem.

Please use the newtype encoding that I used for `IO.Par` in this PR: https://github.com/typelevel/cats-effect/pull/115/

Reply:
To make it clear:

- add `monix.execution.internal.Newtype1` (private to Monix, I hope will standardize on something in the community)
- add `object CombineObservable` like you are doing here, but extend it from `Newtype1` and so the `newtype` will be `CombineObservable.Type[+A]` (and also, there won't be a corresponding `class`)
- on this object add `apply` and `unwrap` functions that work with `.asInstanceOf`

Reply:
@alexandru 
Okay, I implemented something that works but I think it could be nice to organize it into something like what you did in you PR with ```IO.Par``` instead of ```CombineObservable.Type```.

Also I have followed your guidelines but is there any reason against implementing ```Newtype1``` ( based on your ```IONewtype``` ) as:
``` scala
private[monix] abstract class Newtype1[F[_]] { self =>
  type Base
  trait Tag extends Any
  type Type[+A] <: Base with Tag

  def apply[A](fa: F[A]): Type[A] =
    fa.asInstanceOf[Type[A]]

  def unwrap[A](fa: Type[A]): F[A] =
    fa.asInstanceOf[F[A]]
}
```

Reply:
Right, if it works, it's fine. Note that's probably an `F[+_]` in the case of `Observable`, but then I don't see a reason for why invariance wouldn't work here.

Reply:
It actually works with just ```F[_]``` for this use case